tests/sync/test_syncing.py

```diff
@@ -7,32 +8,76 @@
 from kitovu import utils
 from kitovu.sync import syncing
 from kitovu.sync.plugin import smb
+from kitovu.sync.settings import ConnectionSettings
+from kitovu.sync.syncplugin import AbstractSyncPlugin
 from helpers import dummyplugin
 
 
+class FakePlugin(AbstractSyncPlugin):
+    def configure(self, info: typing.Dict[str, typing.Any]) -> None:
+        raise NotImplementedError
+
+    def connect(self) -> None:
+        raise NotImplementedError
+
+    def disconnect(self) -> None:
+        raise NotImplementedError
+
+    def create_local_digest(self, path: pathlib.Path) -> str:
+        raise NotImplementedError
+
+    def create_remote_digest(self, path: pathlib.PurePath) -> str:
+        raise NotImplementedError
+
+    def list_path(self, path: pathlib.PurePath) -> typing.Iterable[pathlib.PurePath]:
+        raise NotImplementedError
+
+    def retrieve_file(self, path: pathlib.PurePath, fileobj: typing.IO[bytes]) -> None:
+        raise NotImplementedError
+
+    def connection_schema(self) -> typing.Dict[str, typing.Any]:
+        return {
+            'type': 'object',
+            'properties': {
+                'some-required-prop': {'type': 'string'},
+                'some-optional-prop': {'type': 'string'},
+            },
+            'required': ['some-required-prop'],
+            'allowAdditional': False,
+        }
+
+
 class TestFindPlugin:
 
     def test_find_plugin_builtin(self):
-        plugin = syncing._find_plugin('smb')
+        plugin = syncing._find_plugin(self._get_settings('smb', connection={'username': 'test'}))
         assert isinstance(plugin, smb.SmbPlugin)
 
     def test_find_plugin_missing_external(self, mocker):
         mocker.patch('stevedore.driver.DriverManager', autospec=True,
                      side_effect=stevedore.exception.NoMatches)
 
         with pytest.raises(utils.NoPluginError, match='The plugin doesnotexist was not found'):
-            syncing._find_plugin('doesnotexist')
+            syncing._find_plugin(self._get_settings('doesnotexist'))
 
     def test_find_plugin_external(self, mocker):
-        fake_plugin_obj = object()
+        fake_plugin_obj = FakePlugin()
         manager = mocker.patch('stevedore.driver.DriverManager', autospec=True)
         instance = manager(namespace='kitovu.sync.plugin', name='test',
                            invoke_on_load=True)
         instance.driver = fake_plugin_obj
 
-        plugin = syncing._find_plugin('test')
+        settings = self._get_settings('test', connection={'some-required-prop': 'test'})
+        plugin = syncing._find_plugin(settings)
         assert plugin is fake_plugin_obj
 
+    def _get_settings(self, plugin_name, connection={}, subjects=[]):
```

Comment:
http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments strikes again :wink:

Reply:
Also, this being needed makes the tests more complex (and less like a "microtest") - another good reason why validating the plugin settings should not happen in `_find_plugin` IMHO.

Reply:
WTF? Pyhton doesn't initialize the default arguments for each method call? o.O

I think a plugin shouldn't be able to load if the settings are invalid. But I guess thats a discussion point for the next meeting